qi-doc/scribblings/forms.scrbl

```diff
@@ -14,10 +14,152 @@
+
+The full syntax of Qi ("Standard Qi") is given below. Note that Standard Qi expands to a @seclink["The_Qi_Core_Language"]{smaller core language} before being @seclink["It_s_Languages_All_the_Way_Down"]{compiled to Racket}.
+
+@racketgrammar*[
+[floe _
+      (gen expr ...)
+      △
+      sep
+      ▽
+      collect
+      (esc expr)
+      (clos floe)
+      (as identifier ...)
+      (one-of? expr ...)
+      (all floe)
+      (any floe)
+      (none floe)
+      (and floe ...)
+      (or floe ...)
+      (not floe)
+      (and% floe ...)
+      (or% floe ...)
+      NOT
+      !
+      AND
+      &
+      OR
+      ∥
+      NOR
+      NAND
+      XOR
+      XNOR
+      any?
+      all?
+      none?
+      inverter
+      ⏚
+      ground
+      (~> floe ...)
+      (thread floe ...)
+      (~>> floe ...)
+      (thread-right floe ...)
+      X
+      crossover
+      ==
+      (== floe ...)
+      relay
+      (relay floe ...)
+      (==* floe ...)
+      (relay* floe ...)
+      -<
+      (-< floe ...)
+      tee
+      (tee floe ...)
+      fanout
+      (fanout nat)
+      feedback
+      (feedback nat floe)
+      (feedback nat (then floe) floe)
+      (feedback (while floe) floe)
+      (feedback (while floe) (then floe) floe)
+      count
+      1>
+      2>
+      3>
+      4>
+      5>
+      6>
+      7>
+      8>
+      9>
+      (select index ...)
+      (block index ...)
+      (bundle (index ...) floe floe)
+      group
+      (group nat floe floe)
+      sieve
+      (sieve floe floe floe)
+      (partition [floe floe] ...)
+      (if floe floe)
+      (if floe floe floe)
+      (when floe floe)
+      (unless floe floe)
+      switch
+      (switch switch-expr ...)
+      (switch (% floe) switch-expr ...)
+      (switch (divert floe) switch-expr ...)
+      (gate floe)
+      ><
+      (>< floe)
+      amp
+      (amp floe)
+      pass
+      (pass floe)
+      <<
+      (<< floe)
+      (<< floe floe)
+      >>
+      (>> floe)
+      (>> floe floe)
+      (loop floe)
+      (loop floe floe)
+      (loop floe floe floe)
+      (loop floe floe floe floe)
+      (loop2 floe floe floe)
+      (ε floe floe)
+      (effect floe floe)
+      apply
+      (qi:* expr ...)
+      (expr expr ... __ expr ...)
+      (expr expr ... _ expr ...)
+      (expr expr ...)
+      literal
+      identifier]
+[literal boolean
+         char
+         string
+         bytes
+         number
+         regexp
+         byte-regexp
+         vector-literal
+         box-literal
+         prefab-literal
+         (@#,racket[quote] value)
+         (quasiquote value)
+         (quote-syntax value)
+         (syntax value)]
```

Comment:
Presumably the `@#,racket[quote]` trick avoids turning this into `'value` and lets us put `(quote value)` in the grammar; should we do the same for quasiquote and syntax? (The `quote-syntax` form has no shorthands I'm aware of.)